src/realm/util/file.hpp

```diff
@@ -856,16 +856,14 @@ class File::UnmapGuard {
 /// Only output is supported at this point.
 class File::Streambuf : public std::streambuf {
 public:
-    explicit Streambuf(File*);
+    explicit Streambuf(File*, std::size_t buffer_size = 4096);
```

Comment:
Since we've been hit by a "bug" caused by using an inefficient value, I think it'd be beneficial to leave a brief comment guiding future consumers of the API on pros/cons of small vs large values.

Reply:
@nirinchev, it is a good point, but the following considerations leads me to prefer explaining the performance issue in [here](https://github.com/realm/realm-core/pull/3221#pullrequestreview-195811289) instead:
- `util::File::Streambuf` is intended as a generic utility, and is used for things other than writing Realm files.
- The performance issue appears to be due to an unnaturally expensive write/flush operation.

We can talk about whether 4KiB is the ideal default. I'm fine with raising it to 16KiB.